pdr_backend/pdr_dashboard/util/data.py

```diff
@@ -71,7 +75,8 @@ def get_predictoors_data_from_payouts(
         list: List of processed user payouts stats data.
     """
 
-    for data in user_payout_stats:
+    payout_stats = deepcopy(user_payout_stats)
+    for data in payout_stats:
         formatted_data = format_dict(
```

Comment:
I would suggest format_dict had an extra argument "only_include_keys" which does pick from dict implicitly, and since pick from dict is really simple, we don't need an explicit function, just the functionality in format_dict.

I also think this goes well, logically speaking, into the responsibility of formatting the dict